core/invoker/src/main/scala/whisk/core/container/package.scala

```diff
@@ -57,8 +58,10 @@ package object container {
      * Start time, End time, Some(response) from container consisting of status code and payload
      * If there is no response or an exception, then None.
      */
-    case class RunResult(interval: Interval, response: Option[(Int, String)]) {
+    case class RunResult(interval: Interval, response: Either[ContainerConnectionError, ContainerResponse]) {
         def duration = interval.duration
+        def ok = response.isRight && response.right.get.ok
```

Comment:
equivalent to:

`def ok = response.right.exists(_.ok)`. Yours to decide what is prettier.